// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
// ****************************
// author: wangjingyi
// data : 2020-04-14
// ****************************

syntax = "proto3";
option java_multiple_files = true;
option java_package = "io.grpc.sintol.loginserver";
option java_outer_classname = "LoginServerProto";
option objc_class_prefix = "STL";

package sintol;

//The login service definition.
service LoginServer {
    // send a login logistic
    rpc Login (LoginRequest) returns (LoginReply){}
}

// The request message params contains:
// username: user's login name.
// password: user's login password.
message LoginRequest {
    string username = 1;
    string password = 2;
}

// The reply message params contains:
// retcode : return logicn result code.
// username: user's login name.
// password: user's login password.
// message : it's can return some detail information.
message LoginReply {
    int32 retcode = 1;
    string username = 2;
    string password = 3;
    string message = 4;
}
